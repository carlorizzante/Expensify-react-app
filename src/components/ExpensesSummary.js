import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses.js';
import selectExpensesTotal from '../selectors/expenses-total.js';

export const ExpensesSummary = ({ count, total }) => {
  const label = count > 1 ? "expenses" : "expense";
  const formattedTotal = numeral(total / 100).format('$0,0.00');
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <strong>{ count }</strong> { label } totalling <strong>{ formattedTotal }</strong></h1>
        <div className="page-header__actions">
          <Link className="btn btn-primary" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters);
  const count = expenses.length;
  const total = selectExpensesTotal(expenses);
  return { count, total }
}

export default connect(mapStateToProps)(ExpensesSummary);
