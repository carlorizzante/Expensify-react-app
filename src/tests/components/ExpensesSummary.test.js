import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary.js';

test('should render correctly ExpensesSummary with no expense', () => {
  const wrapper = shallow(<ExpensesSummary count={ 0 } total={ 0 }/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render correctly ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary count={ 1 } total={ 12095 }/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render correctly ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary count={ 3 } total={ 61025 }/>);
  expect(wrapper).toMatchSnapshot();
});
