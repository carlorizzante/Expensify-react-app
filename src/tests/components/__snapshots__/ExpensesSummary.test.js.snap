// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $120.95
  </h1>
</div>
`;

exports[`should render correctly ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $610.25
  </h1>
</div>
`;

exports[`should render correctly ExpensesSummary with no expense 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expense
     totalling 
    $0.00
  </h1>
</div>
`;
